### DISCUSSION TITLE
How to reset a solo2 key?

### DISCUSSION THREAD

**Discussion by burtek** [solokeys/solo2-cli#70] [Q&A]
How so I reset/wipe a solo2 key to remove all resident keys stored on it and regenerate master key?

**Comment by pbl987** [comment]
I would like to know this too. Someone said you can reset it in Win 11 settings menu, but i have Win 10, where i tried to reset it. 
But i suspect it didn't work properly, as i am still having website issues despite newest firmware 2:20220822.0

**Comment by rugk** [comment]
The current [latest firmware release explains how this can be done](https://github.com/solokeys/solo2/releases/tag/2.964.0):

> You can "wipe" a device using fido2-token -R (in the future, the [CLI](https://github.com/solokeys/solo2-cli) will also implement this).

**Reply by rugk** [reply]
Though, in my test (UBS-C hacker edition), that did not work :sweat_smile: 

```
$ fido2-token -L       
/dev/hidraw2: vendor=0x1209, product=0xbeee (SoloKeys Solo 2 Security Key)
$ fido2-token -R /dev/hidraw2
fido2-token: fido_dev_reset: FIDO_ERR_NOT_ALLOWED
```

Installation via `sudo dnf install fido2-tools`  BTW. (Easily found out via `dnf provides fido2-token`.)

**Reply by casphas** [reply]
Same device and error with fido2-token here. Solo2 firmware `2:20220822.0`.

Solo1 cli also gave an error for `solo key reset` and `solo key set-pin`:
`CTAP error: 0x30 - NOT_ALLOWED`
Although this might have been after several unsuccessful attempts to change the PIN, so maybe this was just the key being locked. However this did not change after reinserting the key.

Was able to use Google Chrome Version 115.0.5790.170 to reset the Solo2 key by opening  `chrome://settings/securityKeys` and choosing `Reset your security key`  from the menu.

Setting up a new PIN worked from there as well.

**Comment by manujchandra** [comment]
I have not tried this but you can enter the show URL below in brave and reset the key from there:

![screenshot_2024-05-20_14-43-49](https://github.com/solokeys/solo2-cli/assets/15191493/9c6bad0a-9167-4800-8471-9429ceafbe50)

**Reply by holykol** [reply]
Thanks, that's `chrome://settings/securityKeys` or `brave://settings/securityKeys` or just search for `security keys` in your non-Firefox browser

**Reply by rugk** [reply]
FYI note that Firefox has a similar page called `about:webauthn`, but I did not check whether you can also reset your key there.
